### PR TITLE
Routes controllers

### PR DESCRIPTION
**Pull request by jonathanpeterwu**

### CODE REVIEW

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -35,4 +32,10 @@ def update
 		@item = Item.create(params)
 	end
 
+	private
+
+	def item_params
```
Comment (jonathanpeterwu) [inline app/controllers/items_controller.rb:37]: nice helper this looks great @katherineimogene

File: app/controllers/tabs_controller.rb [hunk 2]
```diff
@@ -11,13 +11,9 @@ def new
...
 		#check parameters of tabs
-		@tab = Tab.new
-		@tab.total = params[:tab][:total]
+		@tab = Tab.new tab_params
 		@tab.date_of_purchase = Time.now
-		@tab.user_id = params[:tab][:user_id]
-		@tab.vendor_id = params[:tab][:vendor_id]
-		@tab.item_list = params[:tab][:item_list]
-		@tab.paid
+		@tab.paid ## Is this supposed to be here like this?
```
Comment (jonathanpeterwu) [inline app/controllers/tabs_controller.rb:16]: @katherineimogene  this should default to false. @tab.paid = false.

where we have our update route we should check for params to see if we can confirm a tab has been paid

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -39,7 +38,24 @@ def edit
...
 
 	def update
-		@user = User.create(params)
+		user = User.find params[:id]
+	end
+
+	def save_coordinates
+		user = User.find params[:id]
+		# TODO - get location from cookie or phone location or ?
+		lat_long = "[47.6097, 122.3331]"
```
Comment (jonathanpeterwu) [inline app/controllers/users_controller.rb:47]: nice this is great, eventually we can hit this route from the mobile app to update routes!

File: app/models/user.rb [hunk 4]
```diff
@@ -1,6 +1,17 @@
...
 	# validates_associated :tabs, :user_ratings
+  def self.by_location location
+    self.select { |u| u.most_recent_location == location }
+  end
+
+  def most_recent_location
+    # TODO. Location translator/geocoder/
+    # Need a model that will translate coordinates to cities,neighborhoods, etc
+    # ideally return something like this here:
+    # Locator.new.city location_history.pop
```
Comment (jonathanpeterwu) [inline app/models/user.rb:14]: i like this thinking. we can put this on the back burner until i figure out how to pass coordinates from the mobile app

File: app/controllers/tabs_controller.rb (hunk 2 above)
Comment (katherineimogene) [inline app/controllers/tabs_controller.rb:16]: @jonathanpeterwu I need to update this controller to mirror users/vendors so we aren't hard-coding params[:tab][:blahblah] for every field. I'll default tab.paid = false (we might have put the default in the db already actually)

**Comment by katherineimogene** [comment]
Merged - cutting a new branch using rails-api, but would like this work to be included.

**Comment by jonathanpeterwu** [comment]
sweet this loooks great

**Pull request merged**